lib/sidebar.rb

```diff
@@ -29,185 +29,193 @@ def pre_render(page)
+        next if page[:title].nil?
+        next if page[:hide_from_sidebar]
 
-        subgroups = merged.select { |subgroup_key, _subgroup_value| subgroup_key.include? key and subgroup_key != key and key != '/' }
+        title = page[:title].split('–').first
+        child_pages = pages.select { |child_path, _| child_path.include?(path) && child_path != path && path != '/' }
 
-        active = active?(filename, key)
-        # puts "#{filename}, #{key}" if active
+        active = filename.active?(path)
+        # puts "#{filename}, #{path}" if active
         item_class = active ? 'nav-group active' : 'nav-group'
 
-        child = "<li class=\"#{item_class}\">"
-        child << "<div class=\"nav-group-heading\"><i class=\"material-icons\">arrow_right</i><span>#{value[:title].split('–').first}</span></div>"
-
-        child << '<ul class="nav-ul">'
-
-        subgroup = generateSubgroup(filename, key, value, subgroups, 2)
-
-        child << subgroup
-
-        if subgroups.any?
-          subgroups.select { |subgroup_key, _subgroup_value| subgroup_key.split('/').length <= 3 }.each do |subgroup_key, subgroup_value|
-            subgroup = generateSubgroup(filename, subgroup_key, subgroup_value, subgroups, 2)
-            child << subgroup
-          end
-        end
-
-        child << '</ul>'
-        child << '</li>'
-        sidebar << child
-      end
-
-      File.open('_site/sidebar.html', 'w') { |f| f.write(sidebar) }
-      sidebar
-    end
-
-    def active?(filename, url, exact = false)
-      if filename == '/' || url == '/'
-        if filename == '/' && url == '/'
-          return true
-        else
-          return false
-        end
-      end
+        child_markup = "<li class=\"#{item_class}\">"
+        child_markup << '<div class="nav-group-heading">'
+        child_markup << '<i class="material-icons">arrow_right</i>'
+        child_markup << "<span>#{title}</span></div>"
 
-      exact ? filename == url : filename.start_with?(url)
-    end
-
-    def merge(hash1, hash2)
-      all_keys = hash1.keys | hash2.keys
-      result_hash = {}
+        child_markup << '<ul class="nav-ul">'
 
-      all_keys.each do |key|
-        hash_value = {}
+        sub_group_markup = generate_sub_group(filename, path, page, child_pages, 2)
 
-        if hash1.key? key
-          value = hash1[key]
+        child_markup << sub_group_markup unless sub_group_markup.nil?
 
-          hash_value = value unless value.nil?
-        end
+        if child_pages.any?
+          child_pages.select { |child_path, _| child_path.split('/').length <= 3 }.each do |child_path, child_page|
+            next if child_page[:title].nil?
+            next if child_page[:hide_from_sidebar]
 
-        if hash2.key? key
-          value = hash2[key]
-
-          hash_value = hash_value.merge(value) unless value.nil?
+            sub_group_markup = generate_sub_group(filename, child_path, child_page, child_pages, 2)
```

Comment:
Codacy found an issue: [Line is too long. [99/80]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)